privacyidea/lib/user.py

```diff
@@ -82,6 +85,12 @@ class User(object):
     realm = ""
     resolver = ""
 
+    # Hash of already checked passwords and their result. If a user has multiple token, it is not necessary to check
+    # the same password multiple times. However, we can not differentiate between PIN+OTP and just PIN, so this dict
+    # will have an entry for PIN+OTP (likely to fail) and then the OTP cut off to just PIN. In case the PIN was
+    # given in the request, the dict will have only one entry.
+    checked_passwords: dict = {}
```

Comment:
I think we should use `_checked_passwords` here to indicate that is a private variable